porch/controllers/remoterootsync/pkg/applyset/comparison.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package applyset
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/klog/v2"
+)
+
+// objectsAreEquals returns true if the two objects are equal.
+// If there is ambiguity / failure, we return false.
+// This method is used to determine whether it is safe to reuse the cached state when replacing objects.
+func objectsAreEqual(l, r *unstructured.Unstructured) bool {
```

Comment:
would reflect.DeepEqual be suitable so we don't need to reimplement the logic?

Reply:
Good call - it's probably good enough - and we end up likely pulling this out into an interface a few commits down anyway!  Removed & replaced with reflect.DeepEqual